src/roswire/ros2/node.py

```diff
@@ -2,28 +2,39 @@
 __all__ = ('ROS2Node',)
 
 import typing
-
 import attr
 
+from ..interface import Node
+from .state import ROS2StateProbe
+
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2Node:
+class ROS2Node(Node):
```

Comment:
The changes here and in node_manager.py don't seem to be related to the topic of the PR. The right call here is to merge the rest of the changes, and then submit these node-related changes in a separate PR.

Reply:
This was actually an accident--I forgot to switch the branches when I started working on the ROS2NodeManager branch.